stock_dropshipping_dual_invoice/wizard/stock_invoice_onshipping.py

```diff
@@ -21,6 +23,23 @@
 class StockInvoiceOnshipping(models.TransientModel):
     _inherit = "stock.invoice.onshipping"
 
+    @api.model
+    def _get_journal_type(self):
+        res_ids = self.env.context.get('active_ids', [])
+        pickings = self.env['stock.picking'].browse(res_ids)
+        pick = pickings and pickings[0]
+        if pick.move_lines:
+            src_usage = pick.move_lines[0].location_id.usage
+            dest_usage = pick.move_lines[0].location_dest_id.usage
+            if src_usage == 'supplier' and dest_usage == 'customer':
+                moves = pick.move_lines.filtered('purchase_line_id')
+                moves = moves and moves[0]
```

Comment:
Maybe also `pick.mapped('move_lines.purchase_line_id.order_id.invoice_method')`? Not sure, but that might be enough.